source/auth/auth.rst

```diff
@@ -683,6 +683,239 @@ mechanism
+           Action=GetCallerIdentity&Version=2011-06-15
+|
+======================== ======================================================================================================
+Name                     Value       
+======================== ======================================================================================================
+HTTP Request Method      POST 
+URI                      /
+Content-Type*            application/x-www-form-urlencoded
+Content-Length*          43
+Host*                    Host field from Server First Message
+Region                   Derived from Host - see below
+X-Amz-Date*              See `Amazon Documentation <https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html?shortFooter=true>`_
+X-Amz-Security-Token*    Optional, See `Amazon Documentation <https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html?shortFooter=true>`_
+X-MongoDB-Server-Nonce*  Base64 string of server nonce
+X-MongoDB-GS2-CB-Flag*   ASCII lower-case character ‘n’ or ‘y’ or ‘p’
+X-MongoDB-Optional-Data* Optional, Optional data, base64 encoded representation of the optional object provided by the client
+Body                     Action=GetCallerIdentity&Version=2011-06-15\n
+======================== ======================================================================================================
+
+.. note::
+        ``*``, Denotes a header that MUST be included in SignedHeaders if present.
+
+.. note::
+        Region is not a header, but simply part of the authorization header. Region by default is ‘us-east-1’ since this is the 
+        implicit region for ‘sts.amazonaws.com’.  Drivers will need to derive the region to use from the endpoint. The region 
+        is the second piece of a FQDN name. While all official AWS STS endpoints start with “sts.”, there are non-AWS hosted 
+        endpoints and test endpoints that will not follow this rule.
+
+================= =========
+Host              Region
+================= =========
+sts.amazonaws.com us-east-1
+first.second      second
+first             us-east-1
+================= ========= 
+
+`MongoCredential`_ Properties
+`````````````````````````````
+
+username
+	MAY be specified. The non-sensitive IAM access key.
+
+source
+	MUST be "$external". Defaults to ``$external``.
+
+password
+	MAY be specified. The sensitive IAM secret key.
+
+mechanism
+	MUST be "MONGODB-IAM"
+
+mechanism_properties
+	AWS_SESSION_TOKEN
+		Drivers MUST allow the user to specify an AWS session token for authentication with temporary credentials.
+
+IAM Credentials
+   If a username and password is provided drivers MUST use these for the IAM access key and IAM secret key, respectively. Drivers
+   MUST use this secret key to generate a signature using HMAC-SHA256.
+
+Temporary Credentials
+   If a username and password is not provided drivers MUST query the standard local AWS endpoint for temporary credentials.
```

Comment:
What does a driver need to do if `username` is set without a `password`? Do we also need to query the local endpoint?

Reply:
Yes

Reply:
Can we state that explicitly? We also need to define the behavior, for example does the driver just ignore the `username` altogether or does it override the temporary AccessKeyId? If we are just going to ignore the `username` when the `password` is not provided I think we should instead raise error.

Reply:
Done. Stated that we must raise an error if just a username or just a password is provided without the other.

Reply:
@ShaneHarvey please comment back or resolve.